src/BudgetManager/BudgetManager.CQRS/Handlers/WalletHandlers/DeleteUserWalletHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using BudgetManager.CQRS.Commands.WalletCommands;
+using BudgetManager.Model;
+using BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation;
+using MediatR;
+using MongoDB.Driver;
+
+namespace BudgetManager.CQRS.Handlers.WalletHandlers
+{
+    public class DeleteUserWalletHandler : IRequestHandler<DeleteUserWalletCommand, bool>
+    {
+        private readonly IBaseRepository<User> _userRepository;
+
+        public DeleteUserWalletHandler(IBaseRepository<User> userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        public async Task<bool> Handle(DeleteUserWalletCommand request, CancellationToken cancellationToken)
+        {
+            var user = await _userRepository.FindByIdAsync(request.userId, cancellationToken);
+            user?.Wallets?.RemoveAll(w => w.Id == request.walletId);
```

Comment:
Is it possible to use Mongo's default methods like elemMatch and PullFilter for removing wallet from user? Because based on tests if we have a lot of wallets then LINQ will require more resources to execute